### PR TITLE
Generalize the Update Resource Status Script

### PR DESCRIPTION
**Pull request by gagelarsen**
**Primary changes in this Pull Request:**

- Adds a generic update Resource status script.
- Adds a generic ResourceCondorWorkflow.

**Please review the checklist before submitting the Pull Request:**

- [x] Pull request has a meaning full name (not just the commit message from of your last commit)
- [x] Code has been linted using flake8
- [x] All methods have accurate Google-Style Docstrings
- [x] 100% test coverage for new content
- [x] Tests for each common use case (please don't write one test that covers all use cases)
- [x] Bonus: Use TypeHints

**Explain deviations from original design if applicable:**

- Combined two tasks together.

### CODE REVIEW

File: tethysext/atcore/resources/job_scripts/update_resource_status.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        # Check Status List
+        if len(status_keys) <= 0:
+            raise ValueError('Argument "status" keys must have at least one status.')
+
+        # Get status for upload keys
+        status_success = True
+        for status_key in status_keys:
+            status = resource.get_status(status_key, None)
+            status_success = status in Resource.OK_STATUSES and status_success
```
Comment (swainn) [inline tethysext/atcore/resources/job_scripts/update_resource_status.py:36]: SMRT!

File: tethysext/atcore/resources/job_scripts/update_resource_status.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+import traceback
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+from tethysext.atcore.models.app_users import Resource
+
+
+def run(resource_db_url: str, resource_id: str, status_keys: list):
+    """
+    Update the status of a Resource
```
Comment (swainn) [inline tethysext/atcore/resources/job_scripts/update_resource_status.py:12]: "Update the root status of a Resource based on the status of one or more other status keys."

File: tethysext/atcore/resources/job_scripts/update_resource_status.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    Args:
+        resource_db_url (str): The url to the resource in the database.
```
Comment (swainn) [inline tethysext/atcore/resources/job_scripts/update_resource_status.py:15]: "The SQLAlchemy URL to the resource database."

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+                 scheduler, job_manager, status_keys, **kwargs):
+        """
+        Constructor.
+
+        Args:
+            user(auth.User): Django user.
+            workflow_name(str): Name of the job.
+            workspace_path(str): Path to workspace to be used by job.
+            resource_db_url(str): SQLAlchemy url to Resource database.
+            resource_id(str): ID of associated resource.
```
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:32]: Missing some args.

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        update_resource_status = CondorWorkflowJobNode(
+            name='finalize',
+            workflow=self.workflow,
+            condorpy_template_name='vanilla_transfer_files',
+            remote_input_files=(
+                update_resource_status_script.__file__,
+            ),
+            attributes=dict(
+                executable='update_resource_status.py',
+                transfer_input_files=(f'../{input_archive_filename}',),
```
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:87]: I don't think you need this archive to run the job. Remove it.

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            job_type='CONDORWORKFLOW',
+            max_jobs={'geoserver': 1},
+            workspace=self.workspace_path,
+            scheduler=self.scheduler
+        )
+        self.workflow.save()
+
+        user_defined_jobs = self.get_jobs()
+
+        input_archive_filename = os.path.basename(self.input_archive_path)
```
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:75]: I don't think you need this.

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+        self.job_name = workflow_name
+        self.safe_job_name = ''.join(s if s.isalnum() else '_' for s in self.job_name)  #: Safe name with only A-Z 0-9
+        self.workspace_path = workspace_path
+        self.resource_db_url = resource_db_url
+        self.resource_id = resource_id
+        self.workflow = None
+        self.scheduler = scheduler
+        self.job_manager = job_manager
+        self.status_keys = status_keys
+        self.input_archive_path = os.path.join(workspace_path, 'condor_upload.zip')
```
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:44]: what is condor_upload.zip? I don't think you need this.

File: tethysext/atcore/resources/job_scripts/update_resource_status.py (hunk 2 above)
Comment (gagelarsen) [reply tethysext/atcore/resources/job_scripts/update_resource_status.py]: Fixed

File: tethysext/atcore/resources/job_scripts/update_resource_status.py (hunk 3 above)
Comment (gagelarsen) [reply tethysext/atcore/resources/job_scripts/update_resource_status.py]: fixed

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 4 above)
Comment (swainn) [reply tethysext/atcore/services/resource_condor_workflow.py:32]: suggestion for status_keys:

status_keys(list): One or more keys of statuses to check to determine resource status. The other jobs must update these statuses to one of the Resource.OK_STATUSES for the resource to be marked as SUCCESS.

File: tethysext/atcore/job_scripts/update_resource_status.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+
+def run(resource_db_url: str, resource_id: str, status_keys: list):
+    """
+    Update the root status of a resource based ont he status of one or more other statuses of the same resource.
+
+    Args:
+        resource_db_url (str): The SQLAlchemy to the resource database.
+        resource_id (str): The resource ID.
+        status_keys (list): A list of status keys to check.
```
Comment (swainn) [inline tethysext/atcore/job_scripts/update_resource_status.py:17]: suggestion for status_keys:

status_keys(list): One or more keys of statuses to check to determine resource status. The other jobs must update these statuses to one of the Resource.OK_STATUSES for the resource to be marked as SUCCESS.

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 5 above)
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:87]: You may need to add the executable to the transfer_input_files for it to be copied to the working directory. Check how this is defined in AGWA.

**Review by swainn — CHANGES_REQUESTED** [review]
Remove the tethysext/__init__.py

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 4 above)
Comment (gagelarsen) [reply tethysext/atcore/services/resource_condor_workflow.py:32]: Sorry, forgot i had to add a few things.

File: tethysext/atcore/services/resource_condor_workflow.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+        Args:
+            user(auth.User): Django user.
+            workflow_name(str): Name of the job.
+            workspace_path(str): Path to workspace to be used by job.
+            resource_db_url(str): SQLAlchemy url to Resource database.
+            resource_id(str): ID of associated resource.
+            scheduler (Scheduler): The condor scheduler for the application
+            job_manager (JobManger): The condor job manager for the application.
+            status_keys (list): A list of resource status keys to be passed to the update status script.
```
Comment (swainn) [inline tethysext/atcore/services/resource_condor_workflow.py:35]: Pick one format not both: var (type) or var(type)

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 5 above)
Comment (gagelarsen) [reply tethysext/atcore/services/resource_condor_workflow.py:90]: This is just how agwa is doing it.

File: tethysext/atcore/services/resource_condor_workflow.py (hunk 9 above)
Comment (gagelarsen) [reply tethysext/atcore/services/resource_condor_workflow.py:35]: var (type)

File: tethysext/atcore/job_scripts/update_resource_status.py (hunk 8 above)
Comment (gagelarsen) [reply tethysext/atcore/job_scripts/update_resource_status.py:17]: looks good

**Review by swainn — CHANGES_REQUESTED** [review]
Clean lint

**Review by swainn — APPROVED** [review]

**Pull request merged**